packages/cozy-mespapiers-lib/src/components/Views/PapersList.jsx

```diff
@@ -40,32 +42,59 @@ const PapersList = () => {
       enabled: !isLoadingFiles
     }
   )
-
   const isLoadingContacts =
     isQueryLoading(contactQueryResult) || contactQueryResult.hasMore
 
-  const isLoading = isLoadingFiles || isLoadingContacts
+  const queryConnector =
+    buildConnectorsQueryByQualificationLabel(currentFileTheme)
+  const { data: connectors, ...connectorsQueryLeft } = useQuery(
+    queryConnector.definition,
+    queryConnector.options
+  )
+  const isConnectorsLoading = isQueryLoading(connectorsQueryLeft)
+  const connector = connectors?.[0]
+  const connectorSlug = connector?.slug
 
-  if (isLoading) {
-    return (
-      <>
-        <PapersListToolbar selectedThemeLabel={selectedThemeLabel} />
-        <Spinner
-          className="u-flex u-flex-justify-center u-mt-2 u-h-5"
-          size="xxlarge"
-        />
-      </>
-    )
-  }
+  const queryAccounts = buildAccountsQueryBySlug(
```

Comment:
Seems like this component has a lot of logic going on, maybe worth a small refactor to decouple the business logic from the view?

Reply:
il n'y a pas vraiment de logique, ce sont 3 queries à la suite, donc c'est un peu verbeux... j'avais essayé de refactor un peu le return par contre, mais je trouve que la solution actuelle est la meilleure